File: tessoku/python/A09.py
h, w, n = map(int, input().split())
x = [[0] * (w + 9) for _ in range(h + 9)]

for _ in range(n):
    a, b, c, d = map(int, input().split())
    x[a][b] += 1
    x[a][d + 1] -= 1
    x[c + 1][b] -= 1
    x[c + 1][d + 1] += 1

for i in range(1, h + 3):
    for j in range(1, w + 3):
        x[i][j] += x[i][j - 1]

for i in range(1, w + 3):
    for j in range(1, h + 3):
        x[j][i] += x[j - 1][i]

for i in range(1, h + 1):
    for j in range(1, w + 1):
        print(x[i][j], end=" ")
    print()

File: tessoku/rust/src/bin/A09.rs
use proconio::input;

fn main() {
    input! {
        h: usize,
        w: usize,
        n: usize,
        q: [(usize, usize, usize, usize); n],
    }

    let mut grid = vec![vec![0; w + 2]; h + 2];
    for (a, b, c, d) in q {
        grid[a][b] += 1;
        grid[c + 1][d + 1] += 1;
        grid[c + 1][b] -= 1;
        grid[a][d + 1] -= 1;
    }

    for i in 1..=h {
        for j in 1..w {
            grid[i][j + 1] += grid[i][j];
        }
    }

    for i in 1..=w {
        for j in 1..h {
            grid[j + 1][i] += grid[j][i];
        }
    }

    for i in 1..=h {
        for j in 1..=w {
            print!("{} ", grid[i][j]);
        }
        println!();
    }
}
